backend/cfg/cfg_regalloc_validate.mli

```diff
@@ -0,0 +1,18 @@
+[@@@ocaml.warning "+a-4-30-40-41-42"]
+
+module Description : sig
+  type t
+
+  val create : Cfg_with_layout.t -> t
+end
+
+module Error : sig
+  type t
+
+  val print : Format.formatter -> t -> unit
+end
+
+val verify :
```

Comment:
One thing I could propose is to 
In cfg_regalloc_utils.ml:
Change `on_fatal` to `on_fail` and `fatal_callback` to `fail_callback`.
Add new function `validation_fail` that will call `fail_callback` and then raise a specific exception.
Then we can change most of the `fatal`s in validator to `validation_fail` then in `verify` we would catch `Validation_error` exception and add it to the returner error type.

Reply:
Can you rename "verify" to "run"? and "verify_exn" to "test"

Reply:
@gretay-js renamed the `verify` and `verify_exn` functions.

@xclerc actually I don't think we need to catch those exceptions because it's about allocator doing something stupid and unexpected and it isn't allocator missallocating.

Reply:
OK, I was not sure it was the case for all uses.